### PR TITLE
Added tests for RandomExtensions  #12 

### PR DESCRIPTION
**Pull request by georgimanov**

### CODE REVIEW

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        [Fact]
+        public void NextBool_ShouldReturnTrue_MoreThanOrEqualTo50Times()
+        {
+            var random = new Random();
+            int counter = 0;
+            int count = 10;
+            for (int i = 0; i < count; i++)
+            {
+                int currentCount = 100;
+                int currentCounter = 0;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: What is this for? You initialize it and increment it, but never use it?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                var nextBools = new bool[currentCount];
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: This is not used anywhere?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                for (int j = 0; j < currentCount; j++)
+                {
+                    if (random.NextBool(0.5))
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: No need to specify 0.5 explicitly - it's the default value.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.RandomExtensions
+{
+    using System;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextBoolTests
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: Missing test case: What happens if the random is null?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+    public class NextCharTests
+    {
+        [Fact]
+        public void NextChar_ShouldReturnChar()
+        {
+            var random = new Random();
+            var result = random.NextChar();
+
+            Assert.IsType<char>(result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: This check is a but pointless? Can you think of a scenario where the NextChar() method wont return a Char? The return type is guaranteed by the method signature.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_CapitalLetter()
+        {
+            char mockedResult = 'A';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: `(int)mockedResult` - you don't need the explicit cast here.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            randomMock.Setup(r => r.NextDouble()).Returns(0.168);
+            char result = randomMock.Object.NextChar();
+
+            Assert.Equal((int)mockedResult, (int)result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: `(int)mockedResult, (int)result` - no need for the casts

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_AlphabeticLower()
+        {
+            char mockedResult = 'a';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Please remove redundant cast.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+            randomMock.Setup(r => r.NextDouble()).Returns(0.51);
+            char result = randomMock.Object.NextChar(MoreDotNet.Models.CharType.AlphabeticLower);
+
+            Assert.Equal((int)mockedResult, (int)result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Please remove redundant cast.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+        [Fact]
+        public void NextChar_ShoudReturnChar_AlphabeticUpper()
+        {
+            char mockedResult = 'A';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
+            randomMock.Setup(r => r.NextDouble()).Returns(0.49);
+            char result = randomMock.Object.NextChar(MoreDotNet.Models.CharType.AlphabeticUpper);
+
+            Assert.Equal((int)mockedResult, (int)result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Please remove redundant cast.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_AlphabeticUpper()
+        {
+            char mockedResult = 'A';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Please remove redundant cast.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+            Assert.Equal((int)mockedResult, (int)result);
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_AlphabeticAny()
+        {
+            char mockedResult = 'A';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Ok, I think you get the idea - please remove redundant casts in every test.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_AlphabeticLower()
+        {
+            char mockedResult = 'a';
+            var randomMock = new Mock<Random>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>(), It.IsAny<int>())).Returns((int)mockedResult);
+            randomMock.Setup(r => r.NextDouble()).Returns(0.51);
+            char result = randomMock.Object.NextChar(MoreDotNet.Models.CharType.AlphabeticLower);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Please use a using statement for this:
using MoreDotNet.Models;

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+        public void NextChar_ShouldReturnChar()
+        {
+            var random = new Random();
+            var result = random.NextChar();
+
+            Assert.IsType<char>(result);
+        }
+
+        [Fact]
+        public void NextChar_ShoudReturnChar_CapitalLetter()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: 1. Please use the following naming convention -> [MethodName]_[TestSetup]_Should[ExpectedResult]
2. Please merge all the NextChar tests into one, using XUnit Data Driven Tests using the [Theroy] and [MemberData] attributes

See: http://stackoverflow.com/questions/22093843/pass-complex-parameters-to-theory 
Note: PropertyData was replaced with MemberData in the newer XUnit versions, but the example is still applicable

For more detailed info, see: http://ikeptwalking.com/writing-data-driven-tests-using-xunit/

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+
+    using Moq;
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextCharTests
+    {
+        [Fact]
+        public void NextChar_ShouldReturnChar()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextCharTests.cs]: Missing test cases:
What happens if the random is null?
What happens if the NextChar(this Random random, CharType mode) is called with an invalid CharType?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 16]
```diff
@@ -0,0 +1,47 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.RandomExtensions
+{
+    using System;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextDateTime
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Missing test cases: 
- What happens if the random is null?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        [Fact]
+        public void NextDateTime_ShouldReturnDateTime()
+        {
+            var random = new Random();
+            var result = random.NextDateTime();
+
+            Assert.IsType<DateTime>(result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Again - can you think of a scenario where the NextDateTime() method will return anything else?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+            Assert.Equal(counter, dates.Distinct().Count());
+        }
+
+        [Fact]
+        public void NextDateTime_ShouldReturnDateTime_BetweenMinAndMaxValue()
+        {
+            var random = new Random();
+            var mivValue = new DateTime(2016, 1, 1, 0, 0, 0);
+            var maxValue = new DateTime(2018, 1, 1, 0, 0, 0);
+            var result = random.NextDateTime(mivValue, maxValue);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Try running this generation a few hundred times to be sure that the randoms generated are always between the limits.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+    public class NextDoubleTests
+    {
+        [Fact]
+        public void NextDouble_ShouldReturnDouble()
+        {
+            var random = new Random();
+            var result = random.NextDouble(0, 10);
+
+            Assert.IsType<double>(result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: As previews methods.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+
+    using Moq;
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextDoubleTests
+    {
+        [Fact]
+        public void NextDouble_ShouldReturnDouble()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: Missing test cases:
- What happens if Random is null?
- NextDouble(this Random random, double minValue, double maxValue) - what happens if minValue is greater than maxValue?
- What happens if the minValue or maxValue are Infinity? And NaN?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 21]
```diff
@@ -0,0 +1,148 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.RandomExtensions
+{
+    using System;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextStringTests
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs:11]: Missing test cases:
- What happens if Random is null?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+
+        [Fact]
+        public void NextString_ShouldReturnString_OfAlphabeticChars()
+        {
+            var random = new Random();
+            var result = random.NextString(OneHundred, MoreDotNet.Models.CharType.AlphabeticAny);
+
+            Assert.IsType<string>(result);
+            Assert.Equal(OneHundred, result.Length);
+            for (int i = 0; i < OneHundred; i++)
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: You can do the whole string string Assert in one go:
`var check = result.Select(x => x.IsLetter()).All(x => x);
                Assert.True(check);`

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            var random = new Random();
+            var result = random.NextString(OneHundred, MoreDotNet.Models.CharType.AlphanumericLower);
+
+            Assert.IsType<string>(result);
+            Assert.Equal(OneHundred, result.Length);
+            for (int i = 0; i < OneHundred; i++)
+            {
+                var currentChar = result[i];
+                Assert.True((currentChar >= 48 && currentChar < 58) || (currentChar >= 97 && currentChar < 123));
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Magic numbers - please extract the check into a meaningful method. Or even write a new extension method?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 24]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            var random = new Random();
+            var result = random.NextString(OneHundred, MoreDotNet.Models.CharType.AlphanumericUpper);
+
+            Assert.IsType<string>(result);
+            Assert.Equal(OneHundred, result.Length);
+            for (int i = 0; i < OneHundred; i++)
+            {
+                var currentChar = result[i];
+                Assert.True((currentChar >= 48 && currentChar < 58) || (currentChar >= 65 && currentChar < 91));
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Magic numbers - please extract the check into a meaningful method.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 25]
```diff
@@ -0,0 +1,148 @@
...
+        public void NextString_ShouldReturnString_OfAnyUnicodeChars()
+        {
+           var random = new Random();
+            var result = random.NextString(OneHundred, MoreDotNet.Models.CharType.AnyUnicode);
+
+            Assert.IsType<string>(result);
+            Assert.Equal(OneHundred, result.Length);
+            for (int i = 0; i < OneHundred; i++)
+            {
+                Assert.True(result[i] >= 0 && result[i] < 65536);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Magic numbers - please extract the check into a meaningful method.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+
+    public class NextTimeSpanTests
+    {
+        [Fact]
+        public void NextTimeSpan_ShouldReturnTimeSpan()
+        {
+            var random = new Random();
+            var result = random.NextTimeSpan();
+
+            Assert.IsType<TimeSpan>(result);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs]: Same as above.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+    using System;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextTimeSpanTests
+    {
+        [Fact]
+        public void NextTimeSpan_ShouldReturnTimeSpan()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs]: Missing test cases:
- What happens if the Random is null?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            var random = new Random();
+            var numbers = new[] { 1, 2, 3, 4, 5, 6 };
+            var number = random.OneOf<int>(numbers);
+
+            Assert.True(numbers.Contains(number));
+        }
+
+        [Fact]
+        public void OneOf_ShouldReturnOneOfIntCollection()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs]: Testing for different types of collections is unnecessary. We are testing our method not the implementations of Arrays, List, etc.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class OneOfTests
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs:10]: Missing test cases:
- What happens if the Random is null?
- What happens if the input list is null?

File: Source/MoreDotNet/Extensions/Common/RandomExtensions.cs [hunk 30]
```diff
@@ -78,7 +78,9 @@ public static bool NextBool(this Random random, double probability = 0.5)
                 throw new ArgumentNullException(nameof(random));
             }
 
-            return random.NextDouble() <= probability;
+            var nextBool = random.NextDouble() <= probability;
```
Comment (Teodor92) [inline Source/MoreDotNet/Extensions/Common/RandomExtensions.cs]: No need to allocate a new variable, when we don't use it anywhere.

File: Source/MoreDotNet/Extensions/Common/RandomExtensions.cs [hunk 31]
```diff
@@ -301,10 +303,14 @@ private static char NextAlphanumericChar(this Random random)
...
             if (numeric)
             {
-                return random.NextNumericChar();
+                var number = random.NextNumericChar();
+
+                return number;
             }
 
-            return random.NextAlphabeticChar(random.NextBool());
+            var aphabeticChar = random.NextAlphabeticChar(random.NextBool());
```
Comment (Teodor92) [inline Source/MoreDotNet/Extensions/Common/RandomExtensions.cs]: No need to allocate a new variable, when we don't use it anywhere.

File: Source/MoreDotNet/Extensions/Common/RandomExtensions.cs [hunk 32]
```diff
@@ -301,10 +303,14 @@ private static char NextAlphanumericChar(this Random random)
 
             if (numeric)
             {
-                return random.NextNumericChar();
+                var number = random.NextNumericChar();
```
Comment (Teodor92) [inline Source/MoreDotNet/Extensions/Common/RandomExtensions.cs]: No need to allocate a new variable, when we don't use it anywhere.

File: Source/MoreDotNet/Extensions/Common/RandomExtensions.cs [hunk 33]
```diff
@@ -314,12 +320,17 @@ private static char NextAlphabeticChar(this Random random, bool uppercase)
                 throw new ArgumentNullException(nameof(random));
             }
 
+            char alphabeticChar;
```
Comment (Teodor92) [inline Source/MoreDotNet/Extensions/Common/RandomExtensions.cs:317]: No need to allocate a new variables, when we don't use them anywhere.

**Review by Teodor92 — CHANGES_REQUESTED** [review]

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 34]
```diff
@@ -9,49 +9,83 @@
...
         [Fact]
-        public void NextDouble_ShouldReturnDouble()
+        public void NextDouble_RandomIsNull_ShouldThrow_NullReferenceException()
         {
-            var random = new Random();
-            var result = random.NextDouble(0, 10);
+            Random random = null;
 
-            Assert.IsType<double>(result);
+            Assert.Throws<NullReferenceException>(() => random.NextDouble());
```
Comment (vladislav-karamfilov) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: Should be `ArgumentNullException`.

**Review by vladislav-karamfilov — CHANGES_REQUESTED** [review]

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 35]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private const int Counter = 10;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: Please use more meaningful names for consts.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 36]
```diff
@@ -0,0 +1,57 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+    using Xunit;
+
+    public class NextDateTime
+    {
+        private const int Counter = 1000;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Please use more meaningful names for consts.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 37]
```diff
@@ -0,0 +1,91 @@
...
+    using System;
+
+    using Moq;
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class NextDoubleTests
+    {
+        private const int Counter = 1000;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: Please use more meaningful names for consts.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs [hunk 38]
```diff
@@ -0,0 +1,61 @@
...
+
+        [Fact]
+        public void NextBool_ShouldHThrow_NullReferenceException()
+        {
+            Random random = null;
+
+            Assert.Throws<NullReferenceException>(() => random.Next());
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextBoolTests.cs]: It should be `random.NextBool()`

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 39]
```diff
@@ -0,0 +1,57 @@
...
+        private DateTime minDate = new DateTime(1999, 10, 31);
+        private DateTime maxDate = new DateTime(2019, 10, 31);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Make the fields read only.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 40]
```diff
@@ -0,0 +1,91 @@
...
+        private const double DoubleResult = 0.15;
+        private const double MinValue = -50;
+        private const double MaxValue = 150;
+
+        [Fact]
+        public void NextDouble_RandomIsNull_ShouldThrow_NullReferenceException()
+        {
+            Random random = null;
+
+            Assert.Throws<NullReferenceException>(() => random.NextDouble());
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: You are testing the system implementation of NextDouble, no the actual overload extension method.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs [hunk 41]
```diff
@@ -0,0 +1,91 @@
...
+                Assert.True(result >= MinValue && result < MaxValue);
+            }
+        }
+
+        [Fact]
+        public void NextDouble_MinValueIsGreaterThanMaxValue_ShouldThrow_ArgumentException()
+        {
+            var random = new Random();
+
+            Assert.Throws<ArgumentException>(() => random.NextDouble(minValue: MaxValue, maxValue: MinValue));
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDoubleTests.cs]: No need for name identifiers.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs [hunk 42]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            Random random = null;
+
+            Assert.Throws<ArgumentNullException>(() => random.NextDateTime());
+        }
+
+        [Fact]
+        public void NextDateTime_MinValueIsGreaterThanMaxValue_ShouldThrow_ArgumentException()
+        {
+            Random random = new Random();
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextDateTime.cs]: Use `var` where possible.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 43]
```diff
@@ -0,0 +1,133 @@
...
+    using System;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+    using MoreDotNet.Wrappers;
+    using Xunit;
+
+    public class NextStringTests
+    {
+        private const int Zero = 0;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Unused const.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 44]
```diff
@@ -0,0 +1,133 @@
...
+
+            Assert.True(check);
+        }
+
+        [Fact]
+        public void NextString_ShouldReturnString_OfAnyUnicodeChars()
+        {
+            var random = new Random();
+            var result = random.NextString(StringLengthLarge, MoreDotNet.Models.CharType.AnyUnicode);
+            var check = result.Select(x => x.GetType() == typeof(char)).All(x => x);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: No need to validate if a string consist of chars - that's in the language specification.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 45]
```diff
@@ -0,0 +1,133 @@
...
+
+            Assert.True(check);
+        }
+
+        [Fact]
+        public void NextString_ShouldThrow_OutOfMemoryException()
+        {
+            var random = new Random();
+            Assert.Throws<OutOfMemoryException>(() => random.NextString(int.MaxValue, MoreDotNet.Models.CharType.AnyUnicode));
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Why should it throw and OutOfMemoryException?

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 46]
```diff
@@ -0,0 +1,133 @@
...
+
+            Assert.True(check);
+        }
+
+        [Fact]
+        public void NextString_ShouldReturnString_OfAlphanumericUpperChars()
+        {
+            var random = new Random();
+            var result = random.NextString(StringLength, MoreDotNet.Models.CharType.AlphanumericUpper);
+            var check = result.Select(x => this.IsAlphanumericUpperChar(x)).All(x => x);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Use method group.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs [hunk 47]
```diff
@@ -0,0 +1,133 @@
...
+
+            Assert.True(check);
+        }
+
+        [Fact]
+        public void NextString_ShouldReturnString_OfAlphanumericLowerChars()
+        {
+            var random = new Random();
+            var result = random.NextString(StringLength, MoreDotNet.Models.CharType.AlphanumericLower);
+            var check = result.Select(x => this.IsAlphanumericLowerChar(x)).All(x => x);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextStringTests.cs]: Use method group.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs [hunk 48]
```diff
@@ -0,0 +1,57 @@
...
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+    using Xunit;
+
+    public class NextTimeSpanTests
+    {
+        private const int Counter = 1000;
+        private TimeSpan minTimeSpan = new TimeSpan(0, 0, 0, 0, 1);
+        private TimeSpan maxTimeSpan = new TimeSpan(1000, 59, 59, 59);
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs]: Make fields read only.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs [hunk 49]
```diff
@@ -0,0 +1,57 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+    using Xunit;
+
+    public class NextTimeSpanTests
+    {
+        private const int Counter = 1000;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/NextTimeSpanTests.cs]: Please use meaningful names for consts.

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs [hunk 50]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        private int[] numbersArr = new[] { 1, 2, 3, 4, 5, 6 };
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs]: - Do not use shorthands.
- Use meaningful names.
- you can do: `private int[] numbersArr = { 1, 2, 3, 4, 5, 6 };`

File: Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs [hunk 51]
```diff
@@ -0,0 +1,40 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.RandomExtensions
+{
+    using System;
+    using System.Collections.Generic;
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs]: Unused using.

**Review by Teodor92 — CHANGES_REQUESTED** [review]

**Review by Teodor92 — APPROVED** [review]

**Review by vladislav-karamfilov — APPROVED** [review]
